x-pack/plugins/elastic_assistant/server/routes/chat/chat_complete_route.ts

```diff
@@ -0,0 +1,237 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { transformError } from '@kbn/securitysolution-es-utils';
+import { Logger } from '@kbn/core/server';
+import {
+  ELASTIC_AI_ASSISTANT_CHAT_COMPLETE_URL,
+  ChatCompleteProps,
+  API_VERSIONS,
+  Message,
+  Replacements,
+  transformRawData,
+  getAnonymizedValue,
+  ConversationResponse,
+} from '@kbn/elastic-assistant-common';
+import { buildRouteValidationWithZod } from '@kbn/elastic-assistant-common/impl/schemas/common';
+import { getRequestAbortedSignal } from '@kbn/data-plugin/server';
+import { ElasticAssistantPluginRouter, GetElser } from '../../types';
+import { buildResponse } from '../../lib/build_response';
+import {
+  DEFAULT_PLUGIN_NAME,
+  appendAssistantMessageToConversation,
+  createOrUpdateConversationWithUserInput,
+  getPluginNameFromRequest,
+  langChainExecute,
+  performChecks,
+} from '../helpers';
+import { transformESSearchToAnonymizationFields } from '../../ai_assistant_data_clients/anonymization_fields/helpers';
+import { EsAnonymizationFieldsSchema } from '../../ai_assistant_data_clients/anonymization_fields/types';
+
+export const SYSTEM_PROMPT_CONTEXT_NON_I18N = (context: string) => {
+  return `CONTEXT:\n"""\n${context}\n"""`;
+};
+
+export const chatCompleteRoute = (
+  router: ElasticAssistantPluginRouter,
+  getElser: GetElser
+): void => {
+  router.versioned
+    .post({
+      access: 'public',
+      path: ELASTIC_AI_ASSISTANT_CHAT_COMPLETE_URL,
+
+      options: {
+        tags: ['access:elasticAssistant'],
+      },
+    })
+    .addVersion(
+      {
+        version: API_VERSIONS.public.v1,
+        validate: {
+          request: {
+            body: buildRouteValidationWithZod(ChatCompleteProps),
+          },
+        },
+      },
+      async (context, request, response) => {
+        const abortSignal = getRequestAbortedSignal(request.events.aborted$);
+        const assistantResponse = buildResponse(response);
+        try {
+          const ctx = await context.resolve(['core', 'elasticAssistant', 'licensing']);
+          const logger: Logger = ctx.elasticAssistant.logger;
+          const telemetry = ctx.elasticAssistant.telemetry;
+
+          // Perform license, authenticated user and FF checks
+          const checkResponse = performChecks({
+            authenticatedUser: true,
+            capability: 'assistantKnowledgeBaseByDefault',
```

Comment:
How do public API's work when behind a feature flag? Like do we need to annotate the OAS so it's not included in any generated public docs? Or is that fine and we just annotate with how to enabled/use the API?